Introduce JDTLS for formatting, add Allure reporting, and improve test stability

FlowerWineMonk opened the pull request:
- Added JDTLS integration to ensure consistent code formatting
- Integrated Allure reporting and refactored for better separation of concerns
- Added test steps and improved wait handling to prevent premature assertions
- Updated .gitignore, cleaned IDE-generated files, and refreshed README
- Replaced outdated annotations and updated dependencies and locators

FlowerWineMonk committed:
Temporary trash

FlowerWineMonk committed:
refactor: introduced jdtls to fix formatting

FlowerWineMonk committed:
feat: add Allure reporting and refactor for separation of concerns

FlowerWineMonk committed:
feat: add steps and introduce jdtls for formatting

FlowerWineMonk committed:
chore: update .gitignore and readme

FlowerWineMonk committed:
chore: remove IDE-generated files from version control

FlowerWineMonk committed:
chore: update .gitignore

FlowerWineMonk committed:
fix: replace annotations

FlowerWineMonk committed:
fix: change wait, locator, and versions

FlowerWineMonk committed:
chore: update .gitignore

FlowerWineMonk committed:
fix: add wait to prevent premature assertions and check alternative URL

FlowerWineMonk committed:
chore: update README.md

Anassstasi commented on src/main/java/vention/customWebElement/WebElementImp.java:
```diff
@@ -31,15 +31,23 @@ private WebElement waitElementToBeVisible(WebElement element) {
+        .until(ExpectedConditions.elementToBeClickable(locator)).click();
+  }
+
+  public static boolean waitForUrlContains(String partialUrl, int seconds) {
```
It's not part of webelement logic, url is not related to any webelement. Let's move it to driver classes

FlowerWineMonk replied:
I will create utils where my waits will be located and called from.

Anassstasi commented on src/main/java/vention/steps/LoginSteps.java:
```diff
@@ -0,0 +1,30 @@
+
+  @Step("Verify home page is visible after login")
+  public boolean isHomePageVisibleAfterLogin() {
+    HomePage homePage = new HomePage();
```
It's not necessary to create a variable if you are using it just once. Feel free to use new HomePage().isOpened()

Anassstasi replied:
BUT are you sure home page object is not going to be used in any other methods in this class in the future? I thin it will.

in this case it make sense to init the needed pages as common thing for the whole class (The comment is related to ALL similar classes (page objects and test clases))

Anassstasi commented on src/main/java/vention/steps/RegistrationSteps.java:
```diff
@@ -0,0 +1,40 @@
+import io.qameta.allure.Step;
+
+public class RegistrationSteps {
+  @Step("Register new user {user.email} with title '{title}' and country code '{countryCode}'")
```
Please separate class defiition and methods with a space line. Relates to all classes

Anassstasi commented on src/main/java/vention/steps/RegistrationSteps.java:
```diff
@@ -0,0 +1,40 @@
+
+public class RegistrationSteps {
+  @Step("Register new user {user.email} with title '{title}' and country code '{countryCode}'")
+  public void registerNewUserFlow(User user, String title, String countryCode) {
```
remove 'flow' from method name. Method name should show action

Anassstasi commented on src/test/java/vention/pagesTest/BaseTest.java:
```diff
@@ -2,19 +2,26 @@
+  protected static final String ANOTHER_OUTBOUND_URL = "/flightList";
   protected static final String FARE_TYPE_ONE_WAY = "oneWay";
-  protected static final String DEPARTURE_DATE = "calendar-day-31/08/2025";
+  protected static final String DEPARTURE_DATE = "calendar-day-30/09/2025";
```
Why is it changed? the previous test dat is no longer valid?
In this case you need to come up with a solution how to manage test data differently, ex. take today's date and add a month to it. There are special utilities for that

Anassstasi commented on src/test/java/vention/pagesTest/BookFlowTest.java:
```diff
@@ -1,36 +1,30 @@
-    homePage.selectDepartureDate(DEPARTURE_DATE);
-    homePage.clickFindFlightsButton();
+    bookingSteps.bookOneWayFlight(FARE_TYPE_ONE_WAY,
+        Cities.NEW_YORK.getCityName(),
```
PLease don't place each argument on a new string line.

FlowerWineMonk replied:
i integrated a formatter and formatter automatically formats my code.

Anassstasi commented on src/main/java/vention/customWebElement/WebElementImp.java:
```diff
@@ -31,15 +31,23 @@ private WebElement waitElementToBeVisible(WebElement element) {
+        .until(ExpectedConditions.visibilityOf(element));
   }
 
   public static void clickByLocator(By locator, int seconds) {
```
1) This method should not be static, it breaks the logic of the class.
The class should represent webelement wrapper but finding elements is logic of page factory. WebElementImpl should have only the methods that can be used directly to an existing webelement

2) let's implement 'format()' method which allows to add locators with '%s' to annotations and formats them. I will send an example 

3) I'd recommend to rename WebElementImpl to smth like 'CustomWebElement' or any similar. We need a name from business logic perspective

Anassstasi reviewed (changes requested):
Check the comments

FlowerWineMonk committed:
feat: improve element handling and fix home page stability

- Move waits to WaitUtils for better reusability
- Add dynamic locator support to CustomWebElement
- Fix home page StaleElementException issues

FlowerWineMonk committed:
refactor: fix Login and Registration related pages and steps
